extensions/mssql/src/contracts.ts

```diff
@@ -36,19 +36,54 @@ export interface RequestSecurityTokenParams {
 	authority: string;
 	provider: string;
 	resource: string;
+	accountId: string;
 	scope: string;
 }
 
 export interface RequestSecurityTokenResponse {
 	accountKey: string;
 	token: string;
+	expiresOn: Number;
 }
 
 export namespace SecurityTokenRequest {
 	export const type = new RequestType<RequestSecurityTokenParams, RequestSecurityTokenResponse, void, void>('account/securityTokenRequest');
 }
 // ------------------------------- </ Security Token Request > ------------------------------------------
 
+// ------------------------------- < Refresh Token Notification > ---------------------------------
+
+/**
+ * Parameters for a refresh token notification sent from STS to ADS
+ */
+export interface RefreshTokenParams {
+	authority: string;
+	provider: string;
+	resource: string;
+	accountId: string;
+	uri: string;
+}
+
+export namespace RefreshTokenNotification {
+	export const type = new NotificationType<RefreshTokenParams, void>('account/refreshToken');
+}
+
+/**
+ * Parameters for a new refresh token sent from ADS to STS
+ */
+export interface TokenRefreshedParams {
+	accountKey: string;
+	token: string;
+	expiresOn: Number;
+	uri: string;
+}
+
+export namespace RefreshToken {
+	export const type = new NotificationType<TokenRefreshedParams, void>('account/tokenRefreshed');
+}
+
+// ------------------------------- </ Refresh Token Request > -------------------------------
```

Comment:
`Refresh Token Notification`

also just to be consistent you should make a separate section for `Token Refreshed`